types/vote.go

```diff
@@ -326,6 +326,23 @@ func (vote *Vote) ToProto() *tmproto.Vote {
 	}
 }
 
+func VotesToProto(votes []*Vote) (res []*tmproto.Vote) {
+	if votes == nil {
+		return nil
+	}
+
+	res = make([]*tmproto.Vote, len(votes))
+	for i, vote := range votes {
+		v := vote.ToProto()
+		// protobuf crashes when serializing "repeated" fields with nil elements
+		if v == nil {
+			v = &tmproto.Vote{}
+		}
+		res[i] = v
+	}
+	return
+}
```

Comment:
As a rule of thumb, try to avoid "naked" returns. It's generally not necessary to name return values unless a `defer` needs them, or it serves some documentation purpose.

```suggestion
func VotesToProto(votes []*Vote) []*tmproto.Vote {
	if votes == nil {
		return nil
	}

	res := make([]*tmproto.Vote, len(votes))
	for i, vote := range votes {
		v := vote.ToProto()
		// protobuf crashes when serializing "repeated" fields with nil elements
		if v == nil {
			v = &tmproto.Vote{}
		}
		res[i] = v
	}
	return res
}
```

Reply:
Thanks for the info. I'll bear it in mind from now on. I didn't pay a lot of attention to these aspects when rebasing this patch: my main concern was getting it to build and pass the tests (albeit, with not much success the first time).